### ISSUE TITLE
For vs2017, amd64_x86 target are not working.

### ISSUE THREAD

**Issue by lygstate** [microsoft/vscode-cmake-tools#1256]
It should use
C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x86
not C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\

The host and target may not be the same
```
d] [87/482  16% :: 8.542] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_mpsc_rb.c.obj
[build] [88/482  16% :: 8.872] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_counters_manager.c.obj
[build] [89/482  16% :: 8.885] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\collections\aeron_int64_to_tagged_ptr_hash_map.c.obj
[build] [90/482  16% :: 9.001] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\collections\aeron_str_to_ptr_hash_map.c.obj
[build] [91/482  17% :: 9.026] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_mpsc_concurrent_array_queue.c.obj
[build] [92/482  17% :: 9.039] Linking C shared library binaries\aeron.dll
[build] FAILED: binaries/aeron.dll lib/aeron.lib 
[build] cmd.exe /C "cmd.exe /C ""C:\Program Files\CMake\bin\cmake.exe" -E __create_def E:\CI-Cor-Ready\xemu-rpc\aeron\build-cmake\Debug\aeron-client\src\main\c\CMakeFiles\aeron.dir\.\exports.def E:\CI-Cor-Ready\xemu-rpc\aeron\build-cmake\Debug\aeron-client\src\main\c\CMakeFiles\aeron.dir\.\exports.def.objs && cd E:\CI-Cor-Ready\xemu-rpc\aeron\build-cmake\Debug" && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=aeron-client\src\main\c\CMakeFiles\aeron.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\link.exe /nologo aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_bit_set.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_counter_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_to_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_to_tagged_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_str_to_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_atomic.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_broadcast_receiver.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_broadcast_transmitter.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_counters_manager.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_distinct_error_log.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_exclusive_term_appender.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_logbuffer_descriptor.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_mpsc_concurrent_array_queue.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_mpsc_rb.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_spsc_concurrent_array_queue.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_spsc_rb.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_appender.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_gap_filler.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_gap_scanner.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_rebuilder.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_scanner.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_unblocker.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_thread.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\protocol\aeron_udp_protocol.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_arrayutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_clock.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_dlopen.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_env.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_error.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_fileutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_http_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_math.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_netutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_parse_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_properties_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_strutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeronc.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_agent.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_alloc.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_client.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_client_conductor.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_context.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_counter.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_exclusive_publication.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_image.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_log_buffer.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_publication.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_socket.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_subscription.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_windows.c.obj  /out:binaries\aeron.dll /implib:lib\aeron.lib /pdb:binaries\aeron.pdb /dll /version:0.0 /machine:x64 /debug /INCREMENTAL  /DEF:aeron-client\src\main\c\CMakeFiles\aeron.dir\.\exports.def  wsock32.lib  ws2_32.lib  Iphlpapi.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
[build] LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\link.exe /nologo aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_bit_set.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_counter_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_to_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_int64_to_tagged_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\collections\aeron_str_to_ptr_hash_map.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_atomic.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_broadcast_receiver.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_broadcast_transmitter.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_counters_manager.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_distinct_error_log.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_exclusive_term_appender.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_logbuffer_descriptor.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_mpsc_concurrent_array_queue.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_mpsc_rb.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_spsc_concurrent_array_queue.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_spsc_rb.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_appender.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_gap_filler.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_gap_scanner.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_rebuilder.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_scanner.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_term_unblocker.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\concurrent\aeron_thread.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\protocol\aeron_udp_protocol.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_arrayutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_clock.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_dlopen.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_env.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_error.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_fileutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_http_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_math.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_netutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_parse_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_properties_util.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\util\aeron_strutil.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeronc.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_agent.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_alloc.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_client.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_client_conductor.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_context.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_counter.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_exclusive_publication.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_image.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_log_buffer.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_publication.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_socket.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_subscription.c.obj aeron-client\src\main\c\CMakeFiles\aeron.dir\aeron_windows.c.obj /out:binaries\aeron.dll /implib:lib\aeron.lib /pdb:binaries\aeron.pdb /dll /version:0.0 /machine:x64 /debug /INCREMENTAL /DEF:aeron-client\src\main\c\CMakeFiles\aeron.dir\.\exports.def wsock32.lib ws2_32.lib Iphlpapi.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:aeron-client\src\main\c\CMakeFiles\aeron.dir/intermediate.manifest aeron-client\src\main\c\CMakeFiles\aeron.dir/manifest.res" failed (exit code 1112) with the following output:
[build] MSVCRTD.lib(chkstk.obj) : fatal error LNK1112: 模块计算机类型“x86”与目标计算机类型“x64”冲突
[build] [92/482  17% :: 9.326] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_exclusive_term_appender.c.obj
[build] [92/482  17% :: 9.349] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_logbuffer_descriptor.c.obj
[build] [92/482  17% :: 9.437] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_term_gap_filler.c.obj
[build] [92/482  18% :: 9.466] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\util\aeron_netutil.c.obj
[build] [92/482  18% :: 9.471] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\util\aeron_http_util.c.obj
[build] [92/482  18% :: 9.482] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_term_appender.c.obj
[build] [92/482  18% :: 9.554] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\util\aeron_fileutil.c.obj
[build] [92/482  18% :: 9.668] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\util\aeron_properties_util.c.obj
[build] [92/482  19% :: 9.792] Building C object aeron-client\src\main\c\CMakeFiles\aeron_static.dir\concurrent\aeron_distinct_error_log.c.obj
[build] ninja: build stopped: subcommand failed.
```

**Comment by andreeis** [comment]
I am seeing x64\x86 toolset being used on my side for all amd64_x86 kits of several VS (2019, 2017, 2015). 
There is another issue we're going to look at later regarding prompter updates of kits changes, configuration settings updates, cache refresh....
Delete the build folder and try again.

**Comment by lygstate** [comment]
tried but still failed

**Comment by andreeis** [comment]
I am surprised that deleting the build folder didn't unblock you. That solved it for all users complaining about this since 1.4.0. Did you also reload your project after that (or close reopen VSCode)?
Set the cmake logging to debug and share the CMake Tools log when you reopen your project after deleting the build directory. Then another log after running "Delete cache and reconfigure" command. Then another log for building your project (either whole via the 'build' command or a simple compile active file (right click on a source).
In the mean time, once we have a fix for what we believe causes this, we can share a vsix to try. These logs would help us not overlook something.

**Comment by lygstate** [comment]
maybe because 1.4.1 not released yet

**Comment by andreeis** [comment]
No it's not released, we are working on it and we estimate an ETA early next week. It includes fixes for regressions from 1.3.1 to 1.4.0.
To help us make sure we don't overlook something, share the logs I described in my reply above and also confirm if you still have trouble after deleting the build folder and reopening VSCode. Maybe you did not reopen your project when you notified us that the workaround does not help.

**Comment by andreeis** [comment]
@lygstate, from your log I see that you are using Ninja. Did you explicitly set the ninja generator in settings? I wonder if you 'inherited' that from 1.3.1, which had a bug to always pick Ninja (if installed) regardless if the settings wouldn't request it and VS having other preferred generators defined. 1.4.0 is picking the right generator but we have another bug that we're fixing now for 1.4.1 release, regarding the generator change not taking effect unless you delete the build folder and reload the project, both actions, nothing in between (even a configure between the delete and reload will make the generator not be considered).

Also another possibility is a partial kit scan, please share us the content of your user kits json file right after migration.

After that, you can try to get unblocked by deleting the build folder, reloading your project and rescanning the kits.

**Comment by lygstate** [comment]
```
[
  {
    "name": "GCC 9.1.0",
    "compilers": {
      "C": "C:\\CI-Tools\\msys64\\usr\\bin\\gcc.exe",
      "CXX": "C:\\CI-Tools\\msys64\\usr\\bin\\g++.exe"
    }
  },
  {
    "name": "GCC 9.2.0",
    "compilers": {
      "C": "C:\\CI-Tools\\msys64\\mingw64\\bin\\gcc.exe",
      "CXX": "C:\\CI-Tools\\msys64\\mingw64\\bin\\g++.exe"
    },
    "preferredGenerator": {
      "name": "MinGW Makefiles"
    },
    "environmentVariables": {
      "CMT_MINGW_PATH": "C:\\CI-Tools\\msys64\\mingw64\\bin"
    }
  },
  {
    "name": "GCC for arm-linux-gnueabihf 4.9.2",
    "compilers": {
      "C": "C:\\MentorGraphics\\linaro\\bin\\arm-linux-gnueabihf-gcc.exe",
      "CXX": "C:\\MentorGraphics\\linaro\\bin\\arm-linux-gnueabihf-g++.exe"
    }
  },
  {
    "name": "GCC for arm-none-eabi 9.2.1",
    "compilers": {
      "C": "C:\\MentorGraphics\\gcc-arm-none-eabi-9-2019-q4-major-win32\\bin\\arm-none-eabi-gcc.exe",
      "CXX": "C:\\MentorGraphics\\gcc-arm-none-eabi-9-2019-q4-major-win32\\bin\\arm-none-eabi-g++.exe"
    }
  },
  {
    "name": "GCC for arm-none-linux-gnueabi 4.8.3",
    "compilers": {
      "C": "C:\\MentorGraphics\\arm-2014.05\\bin\\arm-none-linux-gnueabi-gcc.exe",
      "CXX": "C:\\MentorGraphics\\arm-2014.05\\bin\\arm-none-linux-gnueabi-g++.exe"
    }
  },
  {
    "name": "GCC for i686-w64-mingw32 4.9.2",
    "compilers": {
      "C": "C:\\Strawberry\\c\\bin\\i686-w64-mingw32-gcc.exe",
      "CXX": "C:\\Strawberry\\c\\bin\\i686-w64-mingw32-g++.exe"
    }
  },
  {
    "name": "GCC for x86_64-pc-msys 9.1.0",
    "compilers": {
      "C": "C:\\CI-Tools\\msys64\\usr\\bin\\x86_64-pc-msys-gcc.exe",
      "CXX": "C:\\CI-Tools\\msys64\\usr\\bin\\x86_64-pc-msys-g++.exe"
    }
  },
  {
    "name": "GCC for x86_64-w64-mingw32 9.2.0",
    "compilers": {
      "C": "C:\\CI-Tools\\msys64\\mingw64\\bin\\x86_64-w64-mingw32-gcc.exe",
      "CXX": "C:\\CI-Tools\\msys64\\mingw64\\bin\\x86_64-w64-mingw32-g++.exe"
    },
    "preferredGenerator": {
      "name": "MinGW Makefiles"
    },
    "environmentVariables": {
      "CMT_MINGW_PATH": "C:\\CI-Tools\\msys64\\mingw64\\bin"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - amd64",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - amd64_arm",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "arm",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - amd64_arm64",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "arm64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - amd64_x86",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "win32",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - x86",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - x86_amd64",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - x86_arm",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "arm",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio 15 生成工具 2017 Release - x86_arm64",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "arm64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - amd64",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - amd64_arm",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "arm",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - amd64_arm64",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "arm64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - amd64_x86",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "win32",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - x86",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - x86_amd64",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "x64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - x86_arm",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "arm",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio Community 2019 Release - x86_arm64",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "arm64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "VisualStudio.10.0 - amd64",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 10 2010",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "VisualStudio.10.0 - x86",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 10 2010",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },
  {
    "name": "VisualStudio.10.0 - x86_amd64",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 10 2010",
      "platform": "x64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "VisualStudio.14.0 - amd64",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 14 2015",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "VisualStudio.14.0 - amd64_x86",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 14 2015",
      "platform": "win32",
      "toolset": "host=x64"
    }
  },
  {
    "name": "VisualStudio.14.0 - x86",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 14 2015",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },
  {
    "name": "VisualStudio.14.0 - x86_amd64",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 14 2015",
      "platform": "x64",
      "toolset": "host=x86"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 10.0.0，具有 Visual Studio 15 生成工具 2017 Release (x86)",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 10.0.0，具有 Visual Studio Community 2019 Release (x86)",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 10.0.0，具有 VisualStudio.10.0 (x86)",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 10.0.0，具有 VisualStudio.14.0 (x86)",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 Visual Studio 15 生成工具 2017 Release (amd64)",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "amd64",
    "compilers": {
      "C": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 Visual Studio 15 生成工具 2017 Release (x86)",
    "visualStudio": "d1c3a502",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 Visual Studio Community 2019 Release (amd64)",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "amd64",
    "compilers": {
      "C": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 Visual Studio Community 2019 Release (x86)",
    "visualStudio": "3c73bd68",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 VisualStudio.10.0 (amd64)",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "amd64",
    "compilers": {
      "C": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 VisualStudio.10.0 (x86)",
    "visualStudio": "VisualStudio.10.0",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 VisualStudio.14.0 (amd64)",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "amd64",
    "compilers": {
      "C": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files\\LLVM\\bin\\clang-cl.exe"
    }
  },
  {
    "name": "用于 MSVC 的 Clang 9.0.0，具有 VisualStudio.14.0 (x86)",
    "visualStudio": "VisualStudio.14.0",
    "visualStudioArchitecture": "x86",
    "compilers": {
      "C": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe",
      "CXX": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\Llvm\\bin\\clang-cl.exe"
    }
  }
]
```

**Comment by lygstate** [comment]
I've already re-scan kits.

**Comment by andreeis** [comment]
This should be fixed in CMake Tools 1.4.1 which was published today.
Let us know if you encounter any issues with this release.
